examples/dlint/js.rs

```diff
@@ -0,0 +1,267 @@
+  // TODO(magurotuna): should add some prefix to `code` to prevent from conflicting with builtin
+  // rules
+  stored.insert(code, diagnostics);
+  state.put::<Diagnostics>(stored);
+
+  Ok(serde_json::json!({}))
+}
+
+fn op_add_rule_code(
+  state: &mut OpState,
+  args: Value,
+  _bufs: &mut [ZeroCopyBuf],
+) -> Result<Value, AnyError> {
+  let code_from_js: Code = serde_json::from_value(args).unwrap();
+
+  let mut stored = state.try_take::<Codes>().unwrap_or_else(HashSet::new);
+  stored.insert(code_from_js.code);
+  state.put::<Codes>(stored);
+
+  Ok(serde_json::json!({}))
+}
+
+pub struct JsRuleRunner {
+  runtime: JsRuntime,
+  module_id: i32,
+}
+
+impl JsRuleRunner {
+  /// Create new JsRuntime for running plugin rules.
+  /// If `plugin_paths` is empty, this returns `None`.
+  pub fn new(plugin_paths: &[&str]) -> Option<Box<Self>> {
+    if plugin_paths.is_empty() {
+      return None;
+    }
+
+    let mut runtime = JsRuntime::new(RuntimeOptions {
+      module_loader: Some(Rc::new(FsModuleLoader)),
+      ..Default::default()
+    });
+
+    runtime
+      .execute("visitor.js", include_str!("visitor.js"))
+      .unwrap();
+    runtime.register_op(
+      "add_diagnostics",
+      deno_core::json_op_sync(op_add_diagnostics),
+    );
+    runtime
+      .register_op("add_rule_code", deno_core::json_op_sync(op_add_rule_code));
+
+    // TODO(magurotuna): `futures::executor::block_on` doesn't seem ideal, but works for now
+    let module_id =
+      deno_core::futures::executor::block_on(runtime.load_module(
+        &ModuleSpecifier::resolve_url_or_path("dummy.js").unwrap(),
+        Some(create_dummy_source(plugin_paths)),
+      ))
+      .unwrap();
+
+    Some(Box::new(Self { runtime, module_id }))
+  }
+}
+
+// TODO(magurotuna): FsModuleLoader is copied from:
+// https://github.com/denoland/deno/pull/8381/files#diff-f7e2ff9248fdb8e71463e0858bfa7070680a09d9704db54d678bf86e49fce3e4
+// This feature is going to be added to `deno_core`, then we should delegate to it.
+struct FsModuleLoader;
+
+impl ModuleLoader for FsModuleLoader {
+  fn resolve(
+    &self,
+    _op_state: Rc<RefCell<OpState>>,
+    specifier: &str,
+    referrer: &str,
+    _is_main: bool,
+  ) -> Result<ModuleSpecifier, AnyError> {
+    Ok(ModuleSpecifier::resolve_import(specifier, referrer)?)
+  }
+
+  fn load(
+    &self,
+    _op_state: Rc<RefCell<OpState>>,
+    module_specifier: &ModuleSpecifier,
+    _maybe_referrer: Option<ModuleSpecifier>,
+    _is_dynamic: bool,
+  ) -> Pin<Box<deno_core::ModuleSourceFuture>> {
+    let module_specifier = module_specifier.clone();
+    async move {
+      let path = module_specifier.as_url().to_file_path().unwrap();
+      let content = std::fs::read_to_string(path)?;
+      let module = deno_core::ModuleSource {
+        code: content,
+        module_url_specified: module_specifier.to_string(),
+        module_url_found: module_specifier.to_string(),
+      };
+      Ok(module)
+    }
+    .boxed_local()
+  }
+}
```

Comment:
@magurotuna can you open a PR in main repo to add this loader to `deno_core`?

Reply:
@bartlomieju Sure.